### PR TITLE
Scala 2.13.9

### PR DESCRIPTION
**Pull request by dos65**
Due to https://github.com/scala/scala/pull/9754 it was required to add additional compat method to filter out non relevant type completions.

### CODE REVIEW

File: build.sbt
Comment (kpodsiad) [inline build.sbt:316]: Why, what changed?

File: mtags/src/main/scala-2/scala/meta/internal/pc/CompletionProvider.scala
Comment (dos65) [inline mtags/src/main/scala-2/scala/meta/internal/pc/CompletionProvider.scala:332]: @kpodsiad @tanishiking 
I updated PR description.

This line was added because [this test](https://github.com/scalameta/metals/blob/c2c6b7077f46b9d30e9be3806691e792919bfa92/tests/cross/src/test/scala/tests/pc/CompletionSnippetSuite.scala#L94-L107) and similar to it started failing.

Having the following source:
```
object Main {
   type Bar = List[Int]
   val x: Ba@@
}
```
compiler started returning completion: List(`Bar`, `List`) which is incorrect and might be filtered out by checking over a new field `aliasInfo` in Type and Scope Member

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala:88]: Are we also not matching on `true` ?

File: mtags/src/main/scala-2/scala/meta/internal/pc/CompletionProvider.scala
Comment (tgodzik) [reply mtags/src/main/scala-2/scala/meta/internal/pc/CompletionProvider.scala:332]: `List` should be actually removed due to fuzzy matching, the compiler can return `List` , but it should not be used by Metals

Comment (tgodzik) [reply mtags/src/main/scala-2/scala/meta/internal/pc/CompletionProvider.scala:332]: Ach I see, the compiler gives us the name `MT` instead of List, that seems like a bug really :thinking: 

I am looking if we can avoid adding new directory for Scala 2.13.9

Comment (tgodzik) [reply mtags/src/main/scala-2/scala/meta/internal/pc/CompletionProvider.scala:332]: Alternative I found is:
```
def isAliasType = head.getClass().getMethods().find(_.getName() == "aliasInfo") match {
          case Some(aliasType) => aliasType.invoke(head) match {
            case Some(_) => true
            case _ => false
          }
          case _ => false

        }
```

though that uses reflection :/

**Review by tgodzik — CHANGES_REQUESTED** [review]
Should we maybe instead file a bug, change the test or ignore it for 2.13.9 and fix it in the compiler?

It seems this change https://github.com/scala/scala/pull/9754/files

**Comment by dos65** [comment]
> Should we maybe instead file a bug, change the test or ignore it for 2.13.9 and fix it in the compiler?
> It seems this change https://github.com/scala/scala/pull/9754/files

@tgodzik Maybe. Need to check if it the only case . I'm not fully sure if don't have any additional cases when we are getting non-matching symbols from completions - https://github.com/scalameta/metals-vscode/pull/738

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala:88]:
```suggestion
      case tm: TypeMember  if tm.accessibile =>
```

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala:104]:
```suggestion
      case sm: ScopeMember if sm.accessibile =>
```

**Review by tgodzik — CHANGES_REQUESTED** [review]
We can merge as is to unblock 2.13.9 and we can ponder on the topic later. I would just add the `if tm.accessabile` part before merging

File: tests/unit/src/test/resources/semanticdb/example/Scalalib.scala
Comment (dos65) [inline tests/unit/src/test/resources/semanticdb/example/Scalalib.scala:4]: and this one

File: tests/unit/src/test/resources/definition/example/Scalalib.scala
Comment (dos65) [inline tests/unit/src/test/resources/definition/example/Scalalib.scala:4]: @tgodzik this one looks like an issue. wdyt?

**Comment by dos65** [comment]
@tgodzik I took a look and it seems that the thing with aliases isn't compiler issue.
we use `completionsAt`[https://github.com/scala/scala/blob/2.13.x/src/repl/scala/tools/nsc/interpreter/PresentationCompilation.scala#L122] method from Global while the REPL uses `completionCadidates`[https://github.com/scala/scala/blob/2.13.x/src/repl/scala/tools/nsc/interpreter/PresentationCompilation.scala#L191]. If you look at `completionCadidates` there is some nameFiltering logic.

File: tests/unit/src/test/resources/definition/example/Scalalib.scala
Comment (dos65) [reply tests/unit/src/test/resources/definition/example/Scalalib.scala:4]: Ok, nevermind, it isn't an issue everything is fine

**Comment by tgodzik** [comment]
> @tgodzik I took a look and it seems that the thing with aliases isn't compiler issue. we use [`completionsAt`](https://github.com/scala/scala/blob/2.13.x/src/repl/scala/tools/nsc/interpreter/PresentationCompilation.scala#L122) method from Global while the REPL uses [`completionCadidates`](https://github.com/scala/scala/blob/2.13.x/src/repl/scala/tools/nsc/interpreter/PresentationCompilation.scala#L191). If you look at `completionCandidates` there is some nameFiltering logic.

I think it's an issue with the compiler actually: https://github.com/scala/scala/pull/10148

Name matcher should not supply something that in reality has a different name. The API would be super misleading that way, I think the original fix was a bit of a hack and we should just dealias the type when needed.

**Comment by dos65** [comment]
@tgodzik Let's keep this compat thing at least for a few Netals versions until there will be a new 2.13.x release. I don't think it will be hard to maintain. wdty?

File: build.sbt
Comment (tgodzik) [inline build.sbt:318]:
```suggestion
    if (scalaVersion.value == 2.13.9) {
```
so that we only do it for Scala 2.13.9 and it's fixed later on?

**Review by tgodzik — CHANGES_REQUESTED** [review]
We can merge as is, but I would simplify it a bit since this should only happen for Scala 2.13.9
I raised the PR for upgrading Scala version for debug adapter.

**Review by tgodzik — APPROVED** [review]

**Pull request merged**
